module/bdev/ocf/vbdev_ocf_cache.c

```diff
@@ -0,0 +1,308 @@
+/*   SPDX-License-Identifier: BSD-3-Clause
+ *   Copyright (C) 2025 Huawei Technologies
+ *   All rights reserved.
+ */
+
+#include "vbdev_ocf_cache.h"
+#include "ctx.h"
+#include "utils.h"
+
+struct vbdev_ocf_caches_head g_vbdev_ocf_caches = STAILQ_HEAD_INITIALIZER(g_vbdev_ocf_caches);
+
+int
+vbdev_ocf_cache_create(const char *cache_name, struct vbdev_ocf_cache **out)
+{
+	struct vbdev_ocf_cache *cache;
+	int rc = 0;
+
+	SPDK_DEBUGLOG(vbdev_ocf, "OCF cache '%s': allocating vbdev_ocf_cache and adding it to cache list\n",
+		      cache_name);
+
+	cache = calloc(1, sizeof(struct vbdev_ocf_cache));
+	if (!cache) {
+		SPDK_ERRLOG("OCF cache '%s': failed to allocate memory for vbdev_ocf_cache\n",
+			    cache_name);
+		return -ENOMEM;
+	}
+
+	cache->name = strdup(cache_name);
+	if (!cache->name) {
+		SPDK_ERRLOG("OCF cache '%s': failed to allocate memory for vbdev_ocf_cache name\n",
+			    cache_name);
+		free(cache);
+		return -ENOMEM;
+	}
+
+	STAILQ_INIT(&cache->cores);
```

Comment:
Isn't a separate list for cores redundant as well? OCF already tracks cores for each cache